### PR TITLE
Input desired state doesn't break with static fields

### LINKED ISSUE #11739: Stopping an input with static fields breaks the input
## Expected Behavior

Stopping an input should not permanently damage the input if it has static fields configured.

## Current Behavior

When an input with static fields configured is stopped manually via the UI, the `static_fields` array in MongoDB is corrupted, so that it can't be deserialized again.

Consider the following excerpt from an input in MongoDB:
```
	"static_fields" : [
		{
			"value" : "bar",
			"key" : "foo"
		}
	]
```
When the user clicks on the `Stop input` button on the `system/inputs` page: 
image

The following error is logged in the server log:
```
2021-12-03 14:52:43,280 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
com.mongodb.WriteConcernException: Write failed with error code 2 and error message 'The field 'static_fields' must be an array but is of type object in document {_id: ObjectId('618133e4ff6dba7ad6c84b51')}'
	at com.mongodb.operation.BaseWriteOperation.convertBulkWriteException(BaseWriteOperation.java:194) ~[mongodb-driver-core-3.12.1.jar:?]
	at com.mongodb.operation.BaseWriteOperation.execute(BaseWriteOperation.java:155) ~[mongodb-driver-core-3.12.1.jar:?]
	at com.mongodb.operation.BaseWriteOperation.execute(BaseWriteOperation.java:52) ~[mongodb-driver-core-3.12.1.jar:?]
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:213) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:182) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.executeWriteOperation(DBCollection.java:356) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.update(DBCollection.java:588) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.update(DBCollection.java:507) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.update(DBCollection.java:482) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.update(DBCollection.java:459) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.update(DBCollection.java:527) ~[mongodb-driver-3.12.1.jar:?]
	at com.mongodb.DBCollection.update(DBCollection.java:543) ~[mongodb-driver-3.12.1.jar:?]
	at org.graylog2.database.PersistedServiceImpl.embed(PersistedServiceImpl.java:255) ~[classes/:?]
	at org.graylog2.inputs.InputServiceImpl.addStaticField(InputServiceImpl.java:251) ~[classes/:?]
	at org.graylog2.inputs.PersistedInputsImpl.update(PersistedInputsImpl.java:138) ~[classes/:?]
	at org.graylog2.inputs.PersistedInputsImpl.saveWithoutEvents(PersistedInputsImpl.java:119) ~[classes/:?]
	at org.graylog2.rest.resources.system.inputs.InputStatesResource.persistDesiredState(InputStatesResource.java:172) ~[classes/:?]
	at org.graylog2.rest.resources.system.inputs.InputStatesResource.stop(InputStatesResource.java:134) ~[classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_222]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_222]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_222]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_222]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80) ~[jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255) [jersey-server-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [jersey-common-2.32.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [jersey-server-2.32.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680) [jersey-server-2.32.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [jersey-container-grizzly2-http-2.32.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [grizzly-http-server-2.4.4.jar:2.4.4]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [metrics-core-4.1.9.jar:4.1.9]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_222]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_222]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
```
And the data in MongoDB now looks like this:
```
	"static_fields" : {
		"foo" : "bar"
	}
```

## Possible Solution

When persisting the desired state, we should probably take the same approach at serializing the input like we do when the user reconfigures the input.

## Steps to Reproduce (for bugs)

See "Current behavior" above ☝️ 

## Context

This bug was probably introduced with #11362

## Your Environment

* Graylog Version: `master`
* Java Version:
* Elasticsearch Version:
* MongoDB Version: `5.0`
* Operating System:
* Browser version:

### PR DESCRIPTION
**Pull request by patrickmann**
#11362 introduced a new DB field for inputs, to persist the desired state (running or stopped).
This PR fixes a serialization regression, that was introduced by that change. Specifically, updating the desired state caused static fields to be serialized as an array of key/value, rather than an array of objects.

Resolves #11739

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#3034

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/shared/inputs/PersistedInputs.java [hunk 1]
```diff
@@ -23,6 +23,4 @@
     boolean add(MessageInput e);
     boolean remove(Object o);
     boolean update(String id, MessageInput newInput);
```
Comment (AntonEbel) [inline graylog2-server/src/main/java/org/graylog2/shared/inputs/PersistedInputs.java:25]: If I'm not mistaken, the `add`, `remove` and `update` methods are not used in the code base. I think we should delete them from this interface and the implementation.

**Review by AntonEbel — DISMISSED** [review]

File: graylog2-server/src/main/java/org/graylog2/shared/inputs/PersistedInputs.java (hunk 1 above)
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/shared/inputs/PersistedInputs.java]: You're right. I removed those methods.

**Review by thll — COMMENTED** [review]
LGTM

**Review by thll — APPROVED** [review]

**Comment by patrickmann** [comment]
Did you test this for back-compat with existing Enterprise forwarders? I think the default implementation of desired state was needed to make that work.

**Pull request merged**